File: backend/app/schemas/moment.py
from pydantic import BaseModel #type:ignore
from typing import List, Optional
from datetime import datetime

class CommentSchema(BaseModel):
    user_id: int
    username: str
    content: str
    created_at: str

class LikeSchema(BaseModel):
    user_id: int
    username: str

class MomentResponse(BaseModel):
    id: int
    user_id: int
    username: str
    avatar: Optional[str]
    content: Optional[str]
    images: List[str]
    likes: List[LikeSchema]
    comments: List[CommentSchema]
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/moment.ts
export interface LikeSchema {
  user_id: number;
  username: string;
  avatar?: string;
}

export interface CommentSchema {
  user_id: number;
  username: string;
  avatar?: string;
  content: string;
  created_at: string;
  cid?: string;
}

export interface MomentResponse {
  id: number;
  user_id: number;
  username: string;
  avatar: string | undefined;
  content: string | null;
  images: string[];
  likes: LikeSchema[];
  comments: CommentSchema[];
  created_at: string;
}

export interface MomentCreateData {
  content: string;
  images: string[];
}
